### PR TITLE
Feature: Produce output in inspect findings json format

### PR DESCRIPTION
**Pull request by prabhu**
Scan can produce the result in ShiftLeft inspect json format for inspect users. Following environment variable needs to be set for testing.

- SHIFTLEFT_ORG_ID
- SHIFTLEFT_ORG_TOKEN
- SHIFTLEFT_ACCESS_TOKEN
- SHIFTLEFT_APP

PR also includes azure-pipelines config, plus minor tweaks

### CODE REVIEW

**Comment by prabhu** [comment]
[Sample findings json file](https://github.com/ShiftLeftSecurity/sast-scan/files/4612226/inspect-report.findings.json.txt)

**Comment by prabhu** [comment]
[Sample for a go project](https://github.com/ShiftLeftSecurity/sast-scan/files/4615841/inspect-report.findings.json.txt)

File: lib/inspect.py [hunk 1]
```diff
@@ -236,3 +250,84 @@ def inspect_scan(language, src, reports_dir, convert, repo_context):
...
+                # Iterate through all the runs
+                for run in report_data["runs"]:
+                    try:
+                        rules = {r["id"]: r for r in run["tool"]["driver"]["rules"]}
+                        results = run["results"]
+                        for result in results:
+                            rule = rules.get(result["ruleId"])
+                            for location in result["locations"]:
+                                finding = {
+                                    "id": finding_id,
```
Comment (Preetam) [inline lib/inspect.py:301]: Unnecessary -- assigned by API

File: lib/inspect.py [hunk 2]
```diff
@@ -236,3 +250,84 @@ def inspect_scan(language, src, reports_dir, convert, repo_context):
...
+                                    "app": app_name,
+                                    "type": "oss_scan",
+                                    "title": result["message"]["text"],
+                                    "description": rule["fullDescription"]["text"],
+                                    "internal_id": utils.calculate_line_hash(
```
Comment (Preetam) [inline lib/inspect.py:306]: Needs to be in the `conclusion_id/occurrence_hash` format

File: lib/inspect.py [hunk 3]
```diff
@@ -236,3 +250,84 @@ def inspect_scan(language, src, reports_dir, convert, repo_context):
...
+                                        location["physicalLocation"]["region"][
+                                            "snippet"
+                                        ]["text"]
+                                    ),
+                                    "severity": result["properties"][
```
Comment (Preetam) [inline lib/inspect.py:311]: Needs to be one of `info`, `moderate`, `critical`

File: lib/inspect.py [hunk 4]
```diff
@@ -236,3 +250,84 @@ def inspect_scan(language, src, reports_dir, convert, repo_context):
...
+                                        "issue_severity"
+                                    ].lower(),
+                                    "owasp_category": "",
+                                    "category": result["ruleId"],
+                                    "version_first_seen": "",
```
Comment (Preetam) [inline lib/inspect.py:316]: Unnecessary

File: lib/inspect.py (hunk 2 above)
Comment (Preetam) [reply lib/inspect.py:306]: I suppose it can be `result["ruleId"] + "/" + utils.calculate_line_hash(...)`

**Pull request merged**